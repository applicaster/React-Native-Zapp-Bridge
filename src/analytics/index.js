import { Platform } from 'react-native';
import { prop, assoc, __ } from 'ramda';
import { postEvent } from './analytics';

const getIp = () =>
  fetch('https://api.ipify.org?format=json').then(r => r.json());
const s4 = () =>
  Math.floor((1 + Math.random()) * 0x10000)
    .toString(16)
    .substring(1);
const guid = () => `${s4()}-${s4()}-${s4()}-${s4()}-${s4()}-${s4()}-${s4()}`;

export const sendAnalyticEvent = (
  key,
  properties = {},
  shouldStringifyValue = Platform.select({ android: false, ios: true })
) => {
  /* eslint no-console: 0 */
  let event = {
    key,
    properties,
    id: properties.uuid || guid(),
    timestamp: Math.floor(Date.now() / 1000)
  };
  if (shouldStringifyValue) {
    event = JSON.stringify(event);
  }
  const options = {
    event
  };

  return getIp()
    .then(prop('ip'))
    .then(assoc('ip', __, options))
    .then(postEvent('MorpheusEvent'))
    .catch(console.warn);
};
