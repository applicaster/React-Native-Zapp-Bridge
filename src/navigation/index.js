import { NativeModules, Platform } from 'react-native';

export const closeModalScreen = Platform.select({
  android: () => {
    NativeModules.APReactNativeBridge.handleCommand('stop', {});
  },
  ios: () => {
    NativeModules.ZPReactNativeBridgeListener.postEvent(
      'dismiss_modal_view',
      { animated: 1 },
      () => {}
    );
  }
});
