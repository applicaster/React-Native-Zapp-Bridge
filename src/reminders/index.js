import { NativeModules } from 'react-native';

const warn = method => {
  /* eslint no-console: 0 */
  console.warn(
    `React Native module 'Reminders' not found when trying to call '${method}'`
  );
};

const methodMapping = {
  addReminder: 'addProgramReminder',
  removeReminder: 'removeProgramReminder',
  hasReminder: 'isProgramReminderScheduled'
};

const exports = Object.assign(
  {},
  ...Object.keys(methodMapping).map(k => ({
    [k]: (...args) => {
      const nativeMethodName = methodMapping[k];
      if (NativeModules && NativeModules.Reminders) {
        return NativeModules.Reminders[nativeMethodName](...args);
      }

      warn(nativeMethodName);

      return Promise.reject(
        new Error('failed to find Reminders native module')
      );
    }
  }))
);

export const { addReminder, removeReminder, hasReminder } = exports;
export default { ...exports };
