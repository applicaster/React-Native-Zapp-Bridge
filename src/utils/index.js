import { Dimensions, NativeModules, Platform } from 'react-native';

export const getFrameDims = (maxWidth, ratio = 9 / 16) => {
  const width = maxWidth || Dimensions.get('window').width;
  const getFrameHeight = w => Math.floor(w * ratio);
  return {
    width,
    height: getFrameHeight(width)
  };
};

export const closeModalScreen = Platform.select({
  android: () => {
    NativeModules.APReactNativeBridge.handleCommand('stop', {});
  },
  ios: () => {
    NativeModules.ZPReactNativeBridgeListener.postEvent(
      'dismiss_modal_view',
      { animated: 1 },
      () => {}
    );
  }
});
