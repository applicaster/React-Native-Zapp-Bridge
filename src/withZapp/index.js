import React from 'react';
import { Platform } from 'react-native';

const propParsers = {
  localization: { android: JSON.parse },
  settings: { android: JSON.parse },
  styles: {
    android: (val, props) =>
      JSON.parse(val)[props.is_tablet ? 'tablet' : 'smartphone']
  },
  extra_props: {
    android: val => {
      const extraProps = JSON.parse(val);

      if (
        extraProps.data_source_model &&
        typeof extraProps.data_source_model === 'string'
      ) {
        extraProps.data_source_model = JSON.parse(extraProps.data_source_model);
      }

      return extraProps;
    }
  }
};

const propsParser = (props, platform) =>
  Object.assign(
    {},
    ...Object.keys(props).map(key => ({
      [key]:
        propParsers[key] && propParsers[key][platform]
          ? propParsers[key][platform](props[key], props)
          : props[key]
    }))
  );

export { propsParser };

// withZap is a react native HOC to patch over platform differences between iOS & android
export default WrappedComponent => {
  const ZappedComponent = props => (
    <WrappedComponent {...propsParser(props, Platform.OS)} />
  );

  return ZappedComponent;
};
