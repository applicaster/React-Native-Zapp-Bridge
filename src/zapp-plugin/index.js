import { NativeModules, Platform } from 'react-native';

const { ZappPlugin } = NativeModules;

export default {
  getConfiguration: Platform.select({
    android: pluginId => ZappPlugin.getPluginConfiguration(pluginId),
    ios: pluginId => ZappPlugin.getConfiguration(pluginId)
  })
};
