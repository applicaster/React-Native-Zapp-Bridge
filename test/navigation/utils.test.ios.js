import { NativeModules } from 'react-native';
import { closeModalScreen } from '../../src/navigation';

describe('utils', () => {
  describe('closeModalScreen()', () => {
    let iosCloseMethod;

    beforeEach(() => {
      iosCloseMethod = jest.fn();

      NativeModules.ZPReactNativeBridgeListener = {
        postEvent: iosCloseMethod
      };
    });

    it('calls the correct native method on iOS', () => {
      closeModalScreen();

      expect(iosCloseMethod.mock.calls).toMatchSnapshot();
    });
  });
});
