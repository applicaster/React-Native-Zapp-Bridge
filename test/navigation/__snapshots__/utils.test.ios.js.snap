// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils closeModalScreen() calls the correct native method on iOS 1`] = `
Array [
  Array [
    "dismiss_modal_view",
    Object {
      "animated": 1,
    },
    [Function],
  ],
]
`;
